pkg/kubelet/kubelet_network_linux.go

```diff
@@ -40,21 +40,29 @@ func (kl *Kubelet) initNetworkUtil() {
 	ipv6Primary := kl.nodeIP != nil && utilnet.IsIPv6(kl.nodeIP)
 
 	var iptClients []utiliptables.Interface
+	var protocols []utiliptables.Protocol
 	if maybeDualStack || !ipv6Primary {
+		protocols = append(protocols, utiliptables.ProtocolIPv4)
 		iptClients = append(iptClients, utiliptables.New(exec, utiliptables.ProtocolIPv4))
 	}
 	if maybeDualStack || ipv6Primary {
+		protocols = append(protocols, utiliptables.ProtocolIPv6)
 		iptClients = append(iptClients, utiliptables.New(exec, utiliptables.ProtocolIPv6))
 	}
 
-	for _, iptClient := range iptClients {
-		kl.syncNetworkUtil(iptClient)
-		go iptClient.Monitor(
-			utiliptables.Chain("KUBE-KUBELET-CANARY"),
-			[]utiliptables.Table{utiliptables.TableMangle, utiliptables.TableNAT, utiliptables.TableFilter},
-			func() { kl.syncNetworkUtil(iptClient) },
-			1*time.Minute, wait.NeverStop,
-		)
+	for i := range iptClients {
```

Comment:
`for i, iptClient := iptClients {`

Reply:
that was the bug, you have to capture the variable otherwise the closure function used in the `Monitor` take its reference, and in the next iteration changes it ... as result in a dual stack cluster, if IPv6 is the second family, the monitor only executes `syncNetworkUtil` for IPv6 ... at least is my understanding